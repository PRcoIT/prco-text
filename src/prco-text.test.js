const prcoText = require("./prco-text").prcoText;

describe("prco-text", () => {
  let response;

  beforeEach(() => {
    response = "";
  });

  describe("invalid options", () => {
    describe("when no options given", () => {
      it("should provide default options and show error and usage", async () => {
        process.argv = ["node", "prco-check-status"];
        response = await prcoText();

        expect(response).toMatchSnapshot();
      });
    });

    describe("when missing config_env_file ", () => {
      it("should show error and usage", async () => {
        process.argv = ["node", "prco-check-status", "-c /foo"];
        response = await prcoText();

        expect(response).toMatchSnapshot();
      });
    });

    describe("when help option given", () => {
      it("should only display usage", async () => {
        process.argv = ["node", "prco-check-status", "-h"];
        response = await prcoText();

        expect(response).toMatchSnapshot();
      });
    });

    describe("when invalid service option (not wis or oneguard)", () => {
      it("should show error and usage", async () => {
        process.argv = ["node", "prco-check-status", "-s foo"];
        response = await prcoText();

        expect(response).toMatchSnapshot();
      });
    });

    describe("when invalid 'from' option", () => {
      it("should show error and usage", async () => {
        process.argv = ["node", "prco-check-status"];
        response = await prcoText();
        expect(response).toMatchSnapshot();

        process.argv = ["node", "prco-check-status", "-f 555"];
        response = await prcoText();
        expect(response).toMatchSnapshot();
      });
    });

    describe("when invalid 'to' option", () => {
      it("should show error and usage", async () => {
        process.argv = ["node", "prco-check-status"];
        response = await prcoText();
        expect(response).toMatchSnapshot();

        process.argv = ["node", "prco-check-status", "-t 555"];
        response = await prcoText();
        expect(response).toMatchSnapshot();
      });
    });

    describe("when invalid 'message' option", () => {
      it("should show error and usage", async () => {
        process.argv = ["node", "prco-check-status"];
        response = await prcoText();
        expect(response).toMatchSnapshot();
      });
    });
  });

  describe("valid options", () => {
    describe("when all options valid", () => {
      it("should show response from server", async () => {
        process.argv = ["node", "prco-check-status", "-t", "+14159353327", "-m", "booya"];
        response = await prcoText();
        response = response.replace(/^sid:.*$/gm, "sid:");
        response = response.replace(/^date_created:.*$/gm, "date_created:");
        expect(response).toMatchSnapshot();
      });
    });
  });
});
