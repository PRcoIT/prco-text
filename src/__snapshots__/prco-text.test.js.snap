// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prco-text invalid options when help option given should only display usage 1`] = `
Object {
  "error": [Error],
  "errorMessage": "

USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {
    "_": Array [],
    "h": "",
    "help": "",
    "s": "signalwire",
    "service": "signalwire",
  },
}
`;

exports[`prco-text invalid options when invalid 'from' option should show error and usage 1`] = `
Object {
  "error": [Error: Missing 'to' option.
],
  "errorMessage": "Missing 'to' option.


USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {
    "_": Array [],
    "from": "+16505297726",
    "s": "signalwire",
    "service": "signalwire",
  },
}
`;

exports[`prco-text invalid options when invalid 'from' option should show error and usage 2`] = `
Object {
  "error": [Error: Missing 'to' option.
],
  "errorMessage": "Missing 'to' option.


USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {
    "_": Array [],
    "f": " 555",
    "from": " 555",
    "s": "signalwire",
    "service": "signalwire",
  },
}
`;

exports[`prco-text invalid options when invalid 'message' option should show error and usage 1`] = `
Object {
  "error": [Error: Missing 'to' option.
],
  "errorMessage": "Missing 'to' option.


USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {
    "_": Array [],
    "from": "+16505297726",
    "s": "signalwire",
    "service": "signalwire",
  },
}
`;

exports[`prco-text invalid options when invalid 'to' option should show error and usage 1`] = `
Object {
  "error": [Error: Missing 'to' option.
],
  "errorMessage": "Missing 'to' option.


USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {
    "_": Array [],
    "from": "+16505297726",
    "s": "signalwire",
    "service": "signalwire",
  },
}
`;

exports[`prco-text invalid options when invalid 'to' option should show error and usage 2`] = `
Object {
  "error": [Error: Missing 'message' option.
],
  "errorMessage": "Missing 'message' option.


USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {
    "_": Array [],
    "from": "+16505297726",
    "s": "signalwire",
    "service": "signalwire",
    "t": " 555",
    "to": " 555",
  },
}
`;

exports[`prco-text invalid options when invalid service option (not wis or oneguard) should show error and usage 1`] = `
Object {
  "error": [Error: Invalid service: use twilio or signalwire.
],
  "errorMessage": "Invalid service: use twilio or signalwire.


USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {
    "_": Array [],
    "s": " foo",
    "service": " foo",
  },
}
`;

exports[`prco-text invalid options when missing config_env_file  should show error and usage 1`] = `
Object {
  "error": [Error: Missing env file:  /foo],
  "errorMessage": "Missing env file:  /foo

USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {},
}
`;

exports[`prco-text invalid options when no options given should provide default options and show error and usage 1`] = `
Object {
  "error": [Error: Missing 'to' option.
],
  "errorMessage": "Missing 'to' option.


USAGE

    prco-text [options]

    OPTIONS

        -c,--config_env_file  location of file containing environment variables
                              defaults to $HOME/protected/prco-text-env
        -s,--service          twilio or signalwire -- default: signalwire
        -f,--from             originating phone number -- default: number in .env file
        -t,--to               destination phone number
        -m,--message          message to be sent
        -h,--help             display usage help

EXAMPLE

    $ prco-text --to 415-222-3333 --message 'hello,
    Your car is ready :)

    PRCO
    415-555-1212'


  ",
  "options": Object {
    "_": Array [],
    "from": "+16505297726",
    "s": "signalwire",
    "service": "signalwire",
  },
}
`;

exports[`prco-text valid options when all options valid should show response from server 1`] = `
"---
service: signalwire
sid:
api_version: 2010-04-01
date_created:
date_sent: null
direction: outbound-api
status: queued
price: null
price_unit: USD
from: +16505297726
to: +14159353327
body: booya

      "
`;
